frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
     end)
 end
 
+local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
+function TouchMenu:_recurse(val, path, text, search_for, depth)
+    depth = depth + 1
+    if depth > MAX_MENU_DEPTH then
+        return
+    end
+    for i,v in pairs(val) do
+        local menu_text = nil
+
+        if type(v) == "table" and (type(i) == "number" or i:find("item")) then
+            local menu_entry = val[i].text or (val[i].text_func and  val[i].text_func())
+            local next_text = menu_entry and (text .. "→ " .. tostring(menu_entry)) or text -- drop sub_item_table
+            TouchMenu:_recurse(val[i], path .. "." .. i, next_text, search_for, depth)
+        elseif i == "text_func" then
+            menu_text = v()
+        elseif i == "text" then
+            menu_text = v
+        end
+
+        if menu_text and Utf8Proc.lowercase(menu_text):find(search_for) then
+            table.insert(TouchMenu.foundMenuItem, {menu_text, path .. "." .. i, text})
+        end
+    end
+end
+
+function TouchMenu:onMenuSearch(search_for)
+    TouchMenu.foundMenuItem = {}
+    for i = 1, #self.tab_item_table do
+        TouchMenu:_recurse(self.tab_item_table[i], i, self.tab_item_table[i].text or "ROOT"..i, search_for, 0)
+    end
+
+--[[
+    for i = 1, #TouchMenu.foundMenuItem do
+        print("xxxxxxx->", i,
+            TouchMenu.foundMenuItem[i][1],
+            TouchMenu.foundMenuItem[i][2],
+            TouchMenu.foundMenuItem[i][3])
+    end
+]]
+end
+
+function TouchMenu:onOpenMenu(path, animate_time_s)
+    local function animate_func()
+        if not animate_time_s or animate_time_s == 0.0 then return end
+        local sleep_us = animate_time_s * 1e6
+        UIManager:forceRePaint()
+        ffiUtil.usleep(sleep_us)
+        UIManager:setDirty(nil, "ui")
+    end
```

Comment:
Tried it, searching for "text", and hitting on "Stix Two Text", a font name. Went into a long animation browsing up to page 16 of my font menu :) that I couldn't interrupt. You should probably allow for that.
Moreover, all my taps trying to interrupt it were processed at the end, in effect closing the menu (well, you made me wait for it, and when I gave up trying to abort it, you make it disappear! :) and triggering a few other actions.